### PR TITLE
Add tenant emergency move to SansShell fdbcli modules

### PR DESCRIPTION
**Pull request by sfc-gh-jfu**
After merge: update permissions list to restrict which group of people can run these commands
Provide wrappers for new fdbcli command of the form `tenant emergency_move <start|switch|finish|abort|status>`
Also add sansshell module

> service FDBMove {
>       rpc FDBMoveDataCopy(FDBMoveDataCopyRequest) returns (FDBMoveDataCopyResponse);
>       rpc FDBMoveDataWait(FDBMoveDataWaitRequest) returns (FDBMoveDataWaitResponse);
> }

 to call and wait for `fdb_move_orchestrator.py`

> [jfu@awsuswest2temptest000032 ~]$ ./sanssh --targets localhost:50042 fdb fdbmovedata copy "/etc/foundationdb/fdb.cluster" "\x02awsuswest2temptest000032\x00\x15\x04\x16\x04\x04" "awsuswest2temptest000032_jfu_dst_cluster_1" "awsuswest2temptest000032_jfu_src_cluster_1"

Run fdbmovedata wait 1987524441901984307 to wait for the process to complete
> [jfu@awsuswest2temptest000032 ~]$ ./sanssh --targets localhost:50042 fdb fdbmovedata copy "/etc/foundationdb/fdb.cluster" "\x02awsuswest2temptest000032\x00\x15\x04\x16\x04\x04" "awsuswest2temptest000032_jfu_dst_cluster_1" "awsuswest2temptest000032_jfu_src_cluster_1"

An existing command is already running on the server.
Run fdbmovedata wait 1987524441901984307 to wait for the process to complete
> [jfu@awsuswest2temptest000032 ~]$ ./sanssh --targets localhost:50042 fdb fdbmovedata copy "/etc/foundationdb/fdb.cluster" "\x02awsuswest2temptest000032\x00\x15\x04\x16\x04\x04" "awsuswest2temptest000032_jfu_dst_cluster_1" "awsuswest2temptest000032_jfu_src_cluster_1"

An existing command is already running on the server.
Run fdbmovedata wait 1987524441901984307 to wait for the process to complete
> [jfu@awsuswest2temptest000032 ~]$ ./sanssh --targets localhost:50042 fdb fdbmovedata wait 1723579472755794943

Command execution failure - rpc error: code = Internal desc = Provided ID 1723579472755794943 does not match stored ID 1987524441901984307
> [jfu@awsuswest2temptest000032 ~]$ ./sanssh --targets localhost:50042 fdb fdbmovedata wait 1987524441901984307

Management Cluster FDBCLI Commands: ['/usr/bin/fdbcli', '-C', '/etc/foundationdb/fdb.cluster', '--tls_certificate_file', '/etc/foundationdb/fdb.pem', '--tls_ca_file', '/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem', '--tls_key_file', '/etc/foundationdb/fdb.pem', '--exec', 'metacluster get awsuswest2temptest000032_jfu_dst_cluster_1 JSON']
....

### CODE REVIEW

File: services/fdb/server/fdbmovedata.go
Comment (sfc-gh-srhodes) [inline services/fdb/server/fdbmovedata.go:36]: Maybe add a link to where this lives in the FDB codebase?

File: services/fdb/server/fdbmovedata.go
Comment (sfc-gh-srhodes) [inline services/fdb/server/fdbmovedata.go:101]: We should probably release this lock when waiting on the command.

If we don't, any other concurrent rpcs for this will hang until the lock is released.

File: services/fdb/server/fdbmovedata.go
Comment (sfc-gh-srhodes) [inline services/fdb/server/fdbmovedata.go:99]: Could you make sure this function works when

- A caller calls it, cancels their call, and calls it again
- Two callers call it in parallel

?

**Review by sfc-gh-srhodes — COMMENTED** [review]
I'm happy with the grpc interface.

Please add a test for the server before we merge this. The logic around the asynchronous command is complex enough to deserve testing.

File: services/fdb/server/fdbcli.go
Comment (sfc-gh-jfu) [reply services/fdb/server/fdbcli.go:1161]: The actual argument to fdbcli needs to have a space because `emergency_move` is a subcommand of the `tenant` umbrella of operations. Will the second option pass both strings with a space in between?

Comment (sfc-gh-srhodes) [reply services/fdb/server/fdbcli.go:1161]: Normally when you type in a command line, the shell takes care of splitting the command on whitespace.

That doesn't happen with golang's os/exec package. Instead, what you have here is the equivalent of passing in `/bin/fdbcli "tenant emergency_move"` on the shell.

File: services/fdb/client/client.go
Comment (sfc-gh-elinardi) [inline services/fdb/client/client.go:3572]: do we need to add (*fdbMoveDataCmd) after `func`? 
i.e. `func (*fdbMoveDataCmd)  GetSubpackage(f *flag.FlagSet) *subcommands.Commander {`

File: services/fdb/client/client.go
Comment (sfc-gh-elinardi) [inline services/fdb/client/client.go:3579]: I think it's a good idea to move this struct closer to where it's being used, which is right before `func (r *fdbMoveDataWaitCmd) Execute`

File: services/fdb/client/client.go
Comment (sfc-gh-elinardi) [inline services/fdb/client/client.go:3619]: Let's mention the default value of `numProcs` here.

File: services/fdb/client/client.go
Comment (sfc-gh-elinardi) [inline services/fdb/client/client.go:3671]: What do you think about adding something like this?
`fmt.Fprintf("Run fdbmovedata wait %d to wait for the process to complete", r.Resp.Id)`

File: services/fdb/server/fdbmovedata.go
Comment (sfc-gh-elinardi) [inline services/fdb/server/fdbmovedata.go:109]: do we need to also reset `s.id`?

File: services/fdb/server/fdbmovedata.go
Comment (sfc-gh-elinardi) [inline services/fdb/server/fdbmovedata.go:170]: if user forgets to run `fdbmovedata wait`, will this get reset?

**Comment by sfc-gh-elinardi** [comment]
Let's add some manual test results to the PR description.
For example, running fdbmovedata copy while server's still processing the previous request should result in server returning the previous request id.

File: services/fdb/server/fdbmovedata.go
Comment (sfc-gh-elinardi) [inline services/fdb/server/fdbmovedata.go:74]: let's also log the existing command path, arguments, and ID

File: services/fdb/server/fdbmovedata.go
Comment (sfc-gh-elinardi) [inline services/fdb/server/fdbmovedata.go:103]: also log id

File: services/fdb/server/fdbmovedata.go
Comment (sfc-gh-elinardi) [inline services/fdb/server/fdbmovedata.go:124]: probably a good idea to add some logging in this function. It'd be helpful for debugging

Comment (sfc-gh-jfu) [reply services/fdb/server/fdbmovedata.go:124]: How do you create a logger without `logger := logr.FromContextOrDiscard(ctx)`? This function does not have `ctx`

Comment (sfc-gh-elinardi) [reply services/fdb/server/fdbmovedata.go:127]: You can get context from stream
```
ctx := stream.Context()
logger := logr.FromContextOrDiscard(ctx)
```

**Review by sfc-gh-elinardi — APPROVED** [review]

**Pull request merged**
